heat/core/logical.py

```diff
@@ -519,6 +519,9 @@ def sanitize_input_type(
     # if one of the tensors is distributed, unsplit/gather it
     if x.split is not None and y.split is None:
         t1 = manipulations.resplit(x, axis=None)
+        local_start = x.comm.chunk(t1.gshape, x.split)[0]
+        local_end = local_start + x.lshape[x.split]
+        y = y[local_start:local_end]
```

Comment:
Functions working with DNDarrays should always get the same arguments on all processes. For local operations, we use the underlying PyTorch tensor accessible through _larray_ or to a lesser extent the _lloc_ methods.